src/heretic/r_plane.c

```diff
@@ -508,12 +510,16 @@ void R_DrawPlanes(void)
             continue;
         }
 
+        swirling = (flattranslation[pl->picnum] == -1); // [crispy] adapt swirl from src/doom to src/heretic
         //
         // regular flat
         //
-        lumpnum = firstflat + flattranslation[pl->picnum];
 
-        tempSource = W_CacheLumpNum(lumpnum, PU_STATIC);
+        if (!swirling)
```

Comment:
One last issue before this can get merged: Please make sure that the `(!swirling)` case contains the pristine original code. This is currently not the case, you removed two lines and added three similar (but different) ones.